gst/nnstreamer/elements/gsttensor_split.c

```diff
@@ -202,10 +202,13 @@ static void
 gst_tensor_split_finalize (GObject * object)
 {
   GstTensorSplit *split;
+  guint i;
 
   split = GST_TENSOR_SPLIT (object);
   gst_tensor_split_remove_src_pads (split);
   g_list_free (split->tensorpick);
+  for (i = 0; i < split->tensorseg->len; i++)
+    g_free (g_array_index (split->tensorseg, tensor_dim *, i));
   g_array_free (split->tensorseg, TRUE);
```

Comment:
As free_segment of g_array_free (split->tensorseg, TRUE) is set to TRUE, elements are automatically freed. right ?

Reply:
with ```g_array_free ( array, TRUE)```, ```tensor_dim``` is freed. but ```tensor_dim[i]``` are not freed.

Reply:
Thank you for the detailed description. :+1:

Reply:
To have tensor_dim freed with g_array_free (array, TRUE), it should've been ```g_array_size_new (FALSE, FALSE, sizeof (tensor_dim), split->numbers)```.

Reply:
Anyway, @jaeyun-jung 's idea looks better.